apps/google-analytics-4/frontend/src/components/main-app/analytics-app/AnalyticsApp.tsx

```diff
@@ -41,30 +57,20 @@ const AnalyticsApp = () => {
     }
 
     fetchData();
-  }, []);
+  }, [api, reportRequestParams]);
 
   useEffect(() => {
-    const sliceByDateRange = (dateRange: DateRangeType): RunReportResponse => {
-      let newRows: Row[] = [];
-      switch (dateRange) {
-        case DateRange.LastDay:
-          newRows = runReportResponse.rows.slice(0, 2);
-          break;
-        case DateRange.LastWeek:
-          newRows = runReportResponse.rows.slice(0, 7);
-          break;
-        case DateRange.LastMonth:
-          newRows = runReportResponse.rows.slice(0, 28);
-          break;
-      }
-      return { ...runReportResponse, rows: [...newRows] };
-    };
     if (runReportResponse.rowCount) {
-      const newData = sliceByDateRange(dateRange);
-      setPageViewData(newData);
+      setStartEndDates(getRangeDates(dateRange));
```

Comment:
If the runReportResponse changes, why do we need to reset the dateRange again? I'm probably missing something.

Reply:
I think you are right some of this useEffect logic can absolutely be consolidated. Will adjust 💯